### PR TITLE
Add timeout rplidar driver if successful scans are not recieved

### PR DESCRIPTION
**Pull request by Boberito25**
Sometimes the rplidar ros driver hangs without publishing.  Now if no scans are received for 5 seconds rplidar_ros kills itself and respawns

### CODE REVIEW

File: src/node.cpp [hunk 1]
```diff
@@ -295,6 +300,7 @@ int main(int argc, char * argv[]) {
                              angle_min, angle_max, 
                              frame_id, distance_factor);
                }
+               time_since_last_scan = ros::Time::now().sec;
```
Comment (po1) [inline src/node.cpp:301]: why do we set this in two different places?

**Comment by po1** [comment]
have you successfully tried this?
my fear is that when it hangs, the main loop might be blocked, and thus you may never hit the timeout check

**Comment by haidai** [comment]
Yeah... Maybe you could try this checking in a separate thread?

**Comment by po1** [comment]
we should have the driver publish diagnostics and the app kill it if the diagnostics stall

**Comment by haidai** [comment]
I was thinking that the diagnostic monitor would be a good place for it too, but that seems a little more involved of an implementation for the demo esp. since Vikram is a bit overloaded at the moment.

But when you have the bandwidth again Vikram, I believe you'll only need to subclass [TopicRateMonitor](https://github.com/mayfieldrobotics/gizmo/blob/develop/app/src/app_bringup/diag.py#L104) and hook it up with roslaunch so that it knows which process to kill. Maybe a good way to do it is to have a TopicRateMonitor subclass take a call back function so that the complicated killing & restarting logic can be configured all in the main `gizmo.py` class instead of being in `diag.py`.

File: src/node.cpp (hunk 1 above)
Comment (Boberito25) [inline src/node.cpp:301]: Woops forgot to delete one fo them

**Comment by Boberito25** [comment]
Okay to answer Paul's first question I tested this by unplugging the rplidar after I had started the app.  The node timed-out, died, and respawned successfully.  I havent been able to replicate the bug where the rplidar driver hangs naturally so I'm not sure if this is a good fix. Do we still want to keep this change or do we want to move all this logic to somewhere around gizmo.py

**Comment by Boberito25** [comment]
Work out proper solution later. This branch is harmless and is on the demo robots.  We can setup a proper fix later.

**Pull request closed without merging**